### PR TITLE
feat(context): add binding.toInjectable shortcut

### PR DESCRIPTION
**Pull request by raymondfeng**
Some developers forget to use `createBindingFromClass` and use `toClass`
or `toProvider` directly. In such cases, the metadata from `@injectable`
are not honored. This shortcut method makes it easy to bind all types of
injectable classes.

```ts
@injectable({scope: BindingScope.SINGLETON, tags: {x: 1}})
      class MyInjectableService {
        constructor(@inject('msg') private _msg: string) {}

        getMessage(): string {
          return 'hello ' + this._msg;
        }
      }
      const serviceBinding = ctx
        .bind('myService')
        .toInjectable(MyInjectableService);
```

Please note that `ctx.bind('myService').toClass(MyInjectableService)` does NOT honor `@injectable`.

Signed-off-by: Raymond Feng <enjoyjava@gmail.com>

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by bradharms** [comment]
What would be the harm in allowing the existing binding methods to honor the @injectable metadata instead of adding a new method?

Our application is built entirely around feature packages and specific artifact folders are only used as an internal organization scheme within those features and are not generally recognized to have any programmatic significance. As a result we don't really get much use out of BootComponent and we just register every artifact explicitly from the component for each feature. From the perspective of our app, having to use a different binding method for classes that have @injectable metadata seems somewhat superfluous.

**Comment by raymondfeng** [comment]
@bradharms We should keep `toClass/toProvider/...` as lower-level APIs that just binds the value without extra metadata. Please note that `toInjectable` calls `toClass/toProvider/...` behind the scenes. 

It's also worth mentioning that `toInjectable` allows classes without `@injectable`. Once this PR is landed, application code can use `toInjectable` for all three flavors of classes.

**Review by bradharms — APPROVED** [review]

**Review by bajtos — APPROVED** [review]
Nice DX improvement 👏🏻 I quickly skimmed through the patch and did not find any obvious problem. I appreciate detailed documentation, both in `Binding.md` and in the tsdoc comment for API docs.

I'll leave it up to Context code owners to review the details and give a final approval.

**Review by jannyHou — APPROVED** [review]
👍

**Pull request merged**
